influxql/select.go

```diff
@@ -460,6 +460,11 @@ func buildExprIterator(expr Expr, ic IteratorCreator, opt IteratorOptions, selec
 			return itr, nil
 		}
 	case *BinaryExpr:
+		// If there was no fill, force null fill to ensure LHS and RHS stay in sync.
+		// (opt is a value, not a reference, so modifications should not affect callers' scope.)
+		if opt.Fill == NoFill {
```

Comment:
I'll take a stab at matching the times in the binary expression iterators. Looks like an ExprIterator can advance the older of its sub-iterators until both sub-iterators have a matching timestamp.